src/pages/[platform]/build-a-backend/auth/advanced-workflows/index.mdx

```diff
@@ -40,20 +40,436 @@ import android0 from '/src/fragments/lib/auth/android/advanced/advanced.mdx';
+const cognitoidentity = new CognitoIdentity({
+  region: '<region-from-config>',
+});
+
+// Note: The custom provider class must implement CredentialsAndIdentityIdProvider
+class CustomCredentialsProvider implements CredentialsAndIdentityIdProvider {
+
+  // Example class member that holds the login information
+  federatedLogin?: {
+    domain: string,
+    token: string
+  };
+
+  // Custom method to load the federated login information
+  loadFederatedLogin(login?: typeof this.federatedLogin) {
+    // You may also persist this by caching if needed
+    this.federatedLogin = login;
+  }
+
+  async getCredentialsAndIdentityId(
+    getCredentialsOptions: GetCredentialsOptions
+  ): Promise<CredentialsAndIdentityId | undefined> {
+    try {
+
+      // You can add in some validation to check if the token is available before proceeding
+      // You can also refresh the token if it's expired before proceeding
+
+      const getIdResult = await cognitoidentity.getId({
+        // Get the identityPoolId from config
+        IdentityPoolId: '<identity-pool-id-from-config>',
+        Logins: { [this.federatedLogin.domain]: this.federatedLogin.token },
+      });
+
+      const cognitoCredentialsResult = await cognitoidentity.getCredentialsForIdentity({
+        IdentityId: getIdResult.IdentityId,
+        Logins: { [this.federatedLogin.domain]: this.federatedLogin.token },
+      });
+
+      const credentials: CredentialsAndIdentityId = {
+        credentials: {
+          accessKeyId: cognitoCredentialsResult.Credentials?.AccessKeyId,
+          secretAccessKey: cognitoCredentialsResult.Credentials?.SecretKey,
+          sessionToken: cognitoCredentialsResult.Credentials?.SessionToken,
+          expiration: cognitoCredentialsResult.Credentials?.Expiration,
+        },
+        identityId: getIdResult.IdentityId,
+      };
+      return credentials;
+    } catch (e) {
+      console.log('Error getting credentials: ', e);
+    }
+  }
+  // Implement this to clear any cached credentials and identityId. This can be called when signing out of the federation service.
+  clearCredentialsAndIdentityId(): void {}
+}
+
+// Create an instance of your custom provider
+const customCredentialsProvider = new CustomCredentialsProvider();
+Amplify.configure(awsconfig, {
+  Auth: {
+    // Supply the custom credentials provider to Amplify
+    credentialsProvider: customCredentialsProvider
+  },
+});
+
+```
+
+<Callout>
+
+**Note:** `Amplify.configure()` takes in two parameters - `BackendResources` and `LibraryConfiguration`. Be sure to ass your CustomCredentialsProvider into the second parameter.
+
+</Callout>
+
+Now that the custom credentials provider is built and supplied to `Amplify.configure`, let's look at how you can use the custom credentials provider to finish federation into Cognito identity pool.
+
+<InlineFilter filters={['react-native']}>
+
+### Facebook Sign-in (React Native - Expo)
+
+```javascript
```

Comment:
Question: can it be `js` instead of `javascript` ?
```suggestion
```js
```

Reply:
Either have always worked for me, but it may be good to have it standardized. @reesscot Any opinions on using `/```javascript` vs `/```js`